src/mam4xx/mo_sethet.hpp

```diff
@@ -0,0 +1,181 @@
+KOKKOS_INLINE_FUNCTION
+void gas_washout(const ThreadTeam &team,
+                 int plev,          // calculate from this level below //in
+                 Real xkgm,         // mass flux on rain drop //in
+                 Real xliq_ik,      // liquid rain water content [gm/m^3] // in
+                 ColumnView xhen_i, // henry's law constant
+                 ColumnView tfld_i, // temperature [K]
+                 ColumnView delz_i, // layer depth about interfaces [cm]  // in
+                 ColumnView xeqca,  // internal variable
+                 ColumnView xca,    // internal variable
+                 ColumnView xgas) { // gas concentration // inout
+  //------------------------------------------------------------------------
+  // calculate gas washout by cloud if not saturated
+  //------------------------------------------------------------------------
+  // FIXME: BAD CONSTANTS
+  Real allca = 0.0; // total of ca between level plev and kk [#/cm3]
+  // ColumnView xca, xeqca;
+  Real const0 = boltz_cgs * 1.0e-6; // [atmospheres/deg k/cm^3]
+  Real geo_fac = 6.0; // geometry factor (surf area/volume = geo_fac/diameter)
+  Real xrm = .189;    // mean diameter of rain drop [cm]
+  Real xum = 748.0;   // mean rain drop terminal velocity [cm/s]
+  //-----------------------------------------------------------------
+  //       ... calculate the saturation concentration eqca
+  //-----------------------------------------------------------------
+  // for (int kk = 0; kk < plev;
+  //     kk++) { // FIXME: not sure if this should be a Kokkos for or not...
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, plev, pver), KOKKOS_LAMBDA(int k) {
+        // cal washout below cloud
+        xeqca(k) = xgas(k) /
+                   (xliq_ik * avo2 + 1.0 / (xhen_i(k) * const0 * tfld_i(k))) *
+                   xliq_ik * avo2;
+        //-----------------------------------------------------------------
+        //       ... calculate ca; inside cloud concentration in  #/cm3(air)
+        //-----------------------------------------------------------------
+        xca(k) = geo_fac * xkgm * xgas(k) / (xrm * xum) * delz_i(k) * xliq_ik *
+                 cm3_2_m3;
+      });
+
+  //-----------------------------------------------------------------
+  //       ... if is not saturated (take hno3 as an example)
+  //               hno3(gas)_new = hno3(gas)_old - hno3(h2o)
+  //           otherwise
+  //               hno3(gas)_new = hno3(gas)_old
+  //-----------------------------------------------------------------
+  for (int kk = 0; kk < plev; kk++) {
+    allca += xca(kk);
+    if (allca < xeqca(kk)) {
+      xgas(kk) = haero::max(xgas(kk) - xca(kk), 0.0);
+    }
+  }
+} // end subroutine gas_washout
+
+//=================================================================================
+KOKKOS_INLINE_FUNCTION
+void find_ktop(
+    Real rlat,        // latitude in radians for columns
```

Comment:
Looking ahead to `sethet` - this rlat value comes from calling [`get_rlat_all_p`](https://github.com/eagles-project/e3sm_mam4_refactor/blob/f6f291b80b9e245a04d422e7bcda5a9609c12107/components/eam/src/physics/cam/phys_grid.F90#L2140). For the C++, should I instead expect `rlat` to come as a parameter from EAMxx?

Reply:
Yes, I think that's reasonable.